handlers/shuttleadapter.go

```diff
@@ -0,0 +1,39 @@
+package handlers
+
+import (
+	"context"
+
+	servicebus "github.com/Azure/azure-service-bus-go"
+	"github.com/Azure/go-shuttle/message"
+	"github.com/devigned/tab"
+)
+
+var _ servicebus.Handler = (*ShuttleAdapter)(nil)
+
+// ShuttleAdapter wraps a go-shuttle message.Handler in a servicebus.Handler.
+// It allows the consumer to write go-shuttle handler func
+// that uses the go-shuttle semantic and message disposition API
+type ShuttleAdapter struct {
+	next message.Handler
+}
+
+func NewShuttleAdapter(next message.Handler) *ShuttleAdapter {
+	return &ShuttleAdapter{next: next}
+}
+
+func (c *ShuttleAdapter) Handle(ctx context.Context, msg *servicebus.Message) error {
+	ctx, s := tab.StartSpan(ctx, "go-shuttle.Listener.HandlerFunc")
+	defer s.End()
+	if c.next == nil {
```

Comment:
you don't want to return an error from a constructor, I can't prevent the user from passing nil. and I don't think you should panic from a ctor either. So that's the best place to validate I think.